src/core/custom-elements/rs-changelog.js

```diff
@@ -0,0 +1,83 @@
+// @ts-check
+/**
+ * Fetches commits between two commitish (tag or commit) - `from` and `to=HEAD`
+ * and shows them as a list of "changes during" those commitish. If a filter
+ * function is provided by the user, it is used to filter the commits that are
+ * to be shown. Otherwise, all commits are shown.
+ *
+ * @typedef {{message: string, hash: string}} Commit
+ */
+import { github } from "../github.js";
+import { hyperHTML } from "../import-maps.js";
+import { pub } from "../pubsubhub.js";
+
+export const name = "rs-changelog";
+
+export default class ChangelogElement extends HTMLElement {
+  constructor() {
+    super();
+    this.props = {
+      from: this.getAttribute("from"),
+      to: this.getAttribute("to") || "HEAD",
+      /** @type {(commit: Commit) => boolean} */
+      filter:
+        typeof window[this.getAttribute("filter")] === "function"
+          ? window[this.getAttribute("filter")]
+          : () => true,
+    };
+  }
+
+  connectedCallback() {
+    const { from, to, filter } = this.props;
+    hyperHTML.bind(this)`
+      <ul>
+      ${{
+        any: fetchCommits(from, to, filter)
+          .then(commits => toHTML(commits))
+          .catch(error => pub("error", error.message))
```

Comment:
oh, we have an element... so we should probably use `showInlineError()` from utils

Reply:
Good point!

Reply:
the element is empty though.. so, we don't get error highlighting :D
can append an error message to element maybe